### PR TITLE
Set up multi jobs for spec testing.

### PR DESCRIPTION
**Pull request by granluo**

### CODE REVIEW

**Comment by granluo** [comment]
No spec changes detected: https://github.com/firebase/firebase-ios-sdk/runs/4670424646?check_suite_focus=true
Test all specs with `skip-tests`: https://github.com/firebase/firebase-ios-sdk/runs/4665958677?check_suite_focus=true

File: scripts/health_metrics/generate_code_coverage_report/Sources/UpdatedFilesCollector/main.swift [hunk 1]
```diff
@@ -70,6 +91,23 @@ struct UpdatedFilesCollector: ParsableCommand {
         if trigger_pod_test_for_coverage_report { break }
       }
     }
+    if let outputPath = outputSDKFileURL {
+      do {
+        // Instead of directly writing Data to a file, trasnfering Data to
```
Comment (paulb777) [inline scripts/health_metrics/generate_code_coverage_report/Sources/UpdatedFilesCollector/main.swift:96]: typo: transferring

File: scripts/health_metrics/code_coverage_file_list.json [hunk 2]
```diff
@@ -75,6 +84,7 @@
   },
   {
     "sdk": "functions",
+    "podspecs": ["FirebaseFunctions.podspec"],
     "filePatterns": [
```
Comment (paulb777) [inline scripts/health_metrics/code_coverage_file_list.json:88]: Add FirebaseFunctionsSwift.podspec

File: .github/workflows/spectesting.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      run: |
+        # The output.json will have podspecs that will be tested.
+        # ``` output.json
+        # [{"podspec":"FirebaseABTesting.podspec"},{"podspec":"FirebaseAnalytics.podspec.json"}]
+        # ```
+        # This array will help generate a matrix for jobs in GHA, taking
+        # advantage of `include`.
+        # https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-including-additional-values-into-combinations
+        # The final matrix will be set as an env var in the GHA workflow,
+        # This env var, matrx, will be passed as a matrix in the next job of
```
Comment (paulb777) [inline .github/workflows/spectesting.yml:32]: typo: matrix

File: .github/workflows/spectesting.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      run: |
+        mkdir specTestingLogs
+        cd ReleaseTooling
+        swift run podspecs-tester --git-root "${GITHUB_WORKSPACE}" --podspec ${PODSPEC} --skip-tests --temp-log-dir "${GITHUB_WORKSPACE}/specTestingLogs"
+    - name: Upload Failed Testing Logs
+      if: failure()
+      uses: actions/upload-artifact@v2
+      with:
+        name: specTestingLogs
+        path: specTestingLogs/*.txt
```
Comment (paulb777) [inline .github/workflows/spectesting.yml:62]: Should the existing `pod spec lint` tests in each product's yml file be removed?

**Review by paulb777 — COMMENTED** [review]
Looking good! Just a few nits and questions

File: scripts/health_metrics/get_updated_files.sh [hunk 5]
```diff
@@ -30,4 +39,9 @@ cd scripts/health_metrics/generate_code_coverage_report
 # merge commit to the head commit from the target branch.
 git diff --name-only remotes/origin/${GITHUB_BASE_REF} ${GITHUB_SHA} > updated_files.txt
 
+if [ -z $spec_output_file] ; then
 swift run UpdatedFilesCollector --changed-file-paths updated_files.txt --code-coverage-file-patterns ../code_coverage_file_list.json
```
Comment (paulb777) [inline scripts/health_metrics/get_updated_files.sh:43]: Indent here and in else

**Review by paulb777 — COMMENTED** [review]
Fix the style and then LGTM

**Review by paulb777 — APPROVED** [review]

**Pull request merged**